packages/address/src/address.ts

```diff
@@ -198,6 +198,26 @@ export default class Address extends AbstractAddress {
     return typeof address === 'string' ? this.fromString(address) : address;
   }
 
+  /**
+   * Takes an ambiguous BECH-32 string or address and creates an `Address`
+   *
+   * @returns a new `Address` instance
+   */
+  static fromAddressOrBech32String(address: string | AbstractAddress): AbstractAddress {
+    if (typeof address === 'string') {
+      if (isBech32(address)) {
+        return new Address(address as Bech32Address);
+      }
+
+      throw new FuelError(
+        FuelError.CODES.INVALID_BECH32_ADDRESS,
+        `Invalid BECH-32 Address: ${address}.`
+      );
+    } else {
+      return address;
+    }
+  }
```

Comment:
```suggestion
  static fromAddressOrBech32String(address: string | AbstractAddress): AbstractAddress {
    if (typeof address !== "string") {
      return address;
    }

    if (!isBech32(address)) {
      throw new FuelError(
        FuelError.CODES.INVALID_BECH32_ADDRESS,
        `Invalid BECH-32 Address: ${address}.`
      );
    }

    return new Address(address as Bech32Address);
  }
```

The suggestion is to reduce nesting, make use of type narrowing, and fail fast.

Reply:
@nedsalk This is better, but I have removed this method entirely since I have discussed it with @arboleya and we agreed that it is better not to enforce the use of the BECH-32 now.